x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/methods/create_custom_rule.ts

```diff
@@ -26,5 +29,10 @@ export const createCustomRule = async (
     data: internalRule,
   });
 
-  return rule;
+  try {
+    return transformValidate(rule);
+  } catch (error) {
+    const zodError = error as ZodError;
```

Comment:
What if it's not a `ZodError`?

Reply:
Thanks! Handled in [`3d37bd6`](https://github.com/elastic/kibana/pull/185748/commits/3d37bd6c45b31abe06685280546a4d93997d64de).